src/crm.ts

```diff
@@ -0,0 +1,113 @@
+import { ActionMessageV2, EventMessageV2 } from "@proca/queue";
+export type handleResult = {
+  pause?: 0; // should it wait delay seconds before reading the next message?
```

Comment:
Not sure if this approach will work.

Generally the RMQ client will feed the messages as they arrive and only keep limit of how many it pre-fetches (it means how many handlers will be run in parallel). You cannot tell it to slow down.

You can add a "retry after some time" in software of course, I used this appraoch to add exponential backoff:
 https://github.com/fixthestatusquo/proca-server/blob/main/sdk/cli/src/service/index.ts#L36
 
 and then i would wrap the handler like so `addBackoff(handler)` and get a new handler.

the syncQueue() expect to get an error thrown if you want to re-deliver the message.

Reply:
should it be the default approach on any error? eg if there is an error, delay and increase the delay the more errors there are?

fyi: the backoff libary hasn't been touched in 7 years

Reply:
removing for now and moving the discussion to https://github.com/fixthestatusquo/proca-sync-template/issues/5